### PR TITLE
fix: Remove 'NT AUTHORITY\Authenticated Users' write permissions on files in c:\k

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

**Reason for Change**:

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

Permissions **before** changes:

```cmd
c:\k NT AUTHORITY\Authenticated Users:(I)(M)
       NT AUTHORITY\Authenticated Users:(I)(OI)(CI)(IO)(M)
       NT AUTHORITY\SYSTEM:(I)(OI)(CI)(F)
       BUILTIN\Administrators:(I)(OI)(CI)(F)
       BUILTIN\Users:(I)(OI)(CI)(RX)

c:\k\nssm.exe NT AUTHORITY\Authenticated Users:(I)(M)
              NT AUTHORITY\SYSTEM:(I)(F)
              BUILTIN\Users:(I)(RX)
              BUILTIN\Administrators:(I)(F)
```

```cmd
Permissions **after** changes:
c:\k BUILTIN\Administrators:(OI)(CI)(F)
       BUILTIN\Users:(OI)(CI)(RX)
       NT AUTHORITY\SYSTEM:(OI)(CI)(F)

c:\k\nssm.exe BUILTIN\Users:(I)(RX)
              BUILTIN\Administrators:(I)(F)
              NT AUTHORITY\SYSTEM:(I)(F)

```

### CODE REVIEW

**Review by AbelHu — DISMISSED** [review]

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -255,6 +255,11 @@ try
         Initialize-DataDirectories
 
         New-Item -ItemType Directory -Path "c:\k" -Force | Out-Null
+        icacls.exe "c:\k" /inheritance:r
```
Comment (AbelHu) [inline parts/k8s/kuberneteswindowssetup.ps1:258]: A log should be helpful to check whether the folder has this fix.

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1:258]: I added some logs.

This was in c:\AzureData\CustomDataSetupScript.log:
image

**Review by AbelHu — APPROVED** [review]

**Review by jsturtevant — APPROVED** [review]

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 2]
```diff
@@ -255,6 +255,13 @@ try
         Initialize-DataDirectories
 
         New-Item -ItemType Directory -Path "c:\k" -Force | Out-Null
+        icacls.exe "c:\k" /inheritance:r
+        icacls.exe "c:\k" /grant:r SYSTEM:`(OI`)`(CI`)`(F`)
+        icacls.exe "c:\k" /grant:r BUILTIN\Administrators:`(OI`)`(CI`)`(F`)
+        icacls.exe "c:\k" /grant:r BUILTIN\Users:`(OI`)`(CI`)`(RX`)
```
Comment (mboersma) [inline parts/k8s/kuberneteswindowssetup.ps1:261]: I infer that the default permissions are less restrictive than what we're setting here.

Is allowing `NT AUTHORITY\Authenticated Users` a general security risk, or is there some specific scenario this addresses? Looks like it does what it says.

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1:261]: Yup, (M) means that user/group can modify files.
kubelet, kube-proxy and other binaries are stored under c:\k so giving non-admins the ability to over-write those files could lead to an escalation of privilege attack.

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**